pallets/moonbeam-foreign-assets/src/lib.rs

```diff
@@ -73,6 +75,54 @@ impl<ForeignAsset> ForeignAssetCreatedHook<ForeignAsset> for () {
 	fn on_asset_created(_foreign_asset: &ForeignAsset, _asset_id: &AssetId) {}
 }
 
+/// Ensure origin location is a sibling
+fn convert_location<T>(location: &Location) -> Result<T::AccountId, DispatchError>
+where
+	T: Config,
+{
+	match location.unpack() {
+		(1, [Parachain(_)]) => T::ConvertLocation::convert_location(location)
+			.ok_or(Error::<T>::CannotConvertLocationToAccount.into()),
+		_ => Err(DispatchError::BadOrigin.into()),
+	}
+}
+#[derive(Decode, Encode, Debug, PartialEq, TypeInfo, Clone)]
+pub enum OriginType {
+	XCM(Location),
+	Governance,
+}
+
+/// Used to convert the success of an EnsureOrigin into `OriginType::Governance`
+pub struct MapSuccessToGovernance<Original>(PhantomData<Original>);
+impl<O, Original: EnsureOrigin<O, Success = ()>> EnsureOrigin<O>
```

Comment:
Implicitly, here it is assumed that having and empty tuple as the Success type, means the Origin comes from either Root or governance, because no account is associated to the origin. While this is true in most cases, it is not strict (we could have a custom Signed origin that returns and empty tuple, and it would be considered Governance)